modules/battenberg/1.0/battenberg.smk

```diff
@@ -0,0 +1,187 @@
+    shell:
+        """
+        R -q -e 'devtools::install_github("Crick-CancerGenomics/ascat/ASCAT")' && ##move some of this to config?
+        R -q -e 'devtools::install_github("morinlab/battenberg")' &&              ##move some of this to config?
+        touch {output.complete}"""
+
+
+# This rule runs the entire Battenberg pipeline. Eventually we may want to set this rule up to allow re-starting
+# of partially completed jobs (e.g. if they run out of RAM and are killed by the cluster, they can automatically retry)
+# TODO: this rule needs to be modified to rely on reference_files and allow setup (downloading) of the Battenberg references
+rule _run_battenberg:
+    input:
+        tumour_bam = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{tumour_id}.bam",
+        tumour_bai = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{tumour_id}.bam.bai",
+        normal_bam = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{normal_id}.bam",
+        normal_bai = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{normal_id}.bam.bai",
+        installed = "config/envs/battenberg_dependencies_installed.success"
+    output:
+        refit=CFG["dirs"]["battenberg"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}/{tumour_id}_refit_suggestion.txt",
+        sub=CFG["dirs"]["battenberg"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}/{tumour_id}_subclones.txt",
+        ac=temp(CFG["dirs"]["battenberg"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}/{tumour_id}_alleleCounts.tab"),
+        mb=temp(CFG["dirs"]["battenberg"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}/{tumour_id}_mutantBAF.tab"),
+        mlrg=temp(CFG["dirs"]["battenberg"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}/{tumour_id}_mutantLogR_gcCorrected.tab"),
+        mlr=temp(CFG["dirs"]["battenberg"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}/{tumour_id}_mutantLogR.tab"),
+        nlr=temp(CFG["dirs"]["battenberg"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}/{tumour_id}_normalLogR.tab"),
+        nb=temp(CFG["dirs"]["battenberg"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}/{tumour_id}_normalBAF.tab")
+    log:
+        stdout = CFG["logs"]["battenberg"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}/{tumour_id}_battenberg.stdout.log",
+        stderr = CFG["logs"]["battenberg"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}/{tumour_id}_battenberg.stderr.log"
+    params:
+        reference_path = lambda w: _battenberg_CFG["reference_path"][w.genome_build],
+        script = CFG["inputs"]["battenberg_script"],
+        calc_sex_status = CFG["inputs"]["calc_sex_status"],
+        x_chrom = lambda w: _battenberg_CFG["options"]["x_chrom"][w.genome_build],
+        y_chrom = lambda w: _battenberg_CFG["options"]["y_chrom"][w.genome_build],
+        chr_prefixed = lambda w: _battenberg_CFG["options"]["chr_prefixed_reference"][w.genome_build],
```

Comment:
I might not be correct, but I think these lambda functions would not handle the hs37d5 reference. Maybe, we can try `op.switch_on_wildcard()` here? In this case, the lambda function here will become, for example, `op.switch_on_wildcard("genome_build", config["switches"]["x_chrom"])`, and in the config file the `options` will be changed to `switches`.

Reply:
I've run this on all three genome builds but it could be that it defaults to a sex (silently) when this fails. We should confirm for sure.

Reply:
I've checked that this is working as expected for the genomes that are handled.